### ISSUE TITLE
[BUG]: iter error when use ptx_coef in ppo:Expected input batch_size (98) to match target batch_size (64).

### ISSUE THREAD

**Issue by baibaiw5** [hpcaitech/ColossalAI#3443]
### 🐛 Describe the bug

It seems there is a bug in the following code:
next(iter(self.pretrain_dataloader)) will create a new batch. The data of input_ids/labels/attention_mask is not aligned with 
each other. They will use differenct batch data. When the batch_size of different batch is different.
An error will occur:Expected input batch_size (98) to match target batch_size (64).

https://github.com/hpcaitech/ColossalAI/blob/main/applications/Chat/coati/trainer/ppo.py
![image](https://user-images.githubusercontent.com/233871/229752309-09513244-19ce-4458-8474-bf1d6ba2869e.png)

![image](https://user-images.githubusercontent.com/233871/229751591-1cbe15b0-4abf-4f0e-ab14-5e9e27e32b85.png)

**Comment by baibaiw5** [comment]
When I change the following code,it run succfully
![image](https://user-images.githubusercontent.com/233871/229755205-33bdca7c-8e39-463a-bd35-fc47805da353.png)

**Comment by yynil** [comment]
https://github.com/hpcaitech/ColossalAI/pull/3433
I just asked for a pr yesterday to fix that

**Comment by RestoreHope2020** [comment]
Thnks wait for it. Hug.

**Comment by HaixHan** [comment]
hi， can you telll me how do you get the prompt dataset when you run the code about the stage3?

**Comment by HaixHan** [comment]
hi， can you tell me how do you get the prompt dataset when you run the code about the stage3?

**Comment by yynil** [comment]
> hi， can you tell me how do you get the prompt dataset when you run the code about the stage3?

Actually all data are just text format and you have to use your own domain data to train your own domain models. You may find the format in my PR codes. The dataset code is the easy_dataset.py.

**Comment by HaixHan** [comment]
ok, I noticed this file in your github code,thans a lot. Besides, i have another question to ask you. In /Chat/coati/trainer/rm.py, the line 104-105, the results of the function model(XX) is a  CausalLMOutputWithCrossAttentions class. So when i run the line 106, it takes error.   Therefore, i think there are some errors in this place. How do you think about it?
















At 2023-04-06 10:34:05, "YY Lin" ***@***.***> wrote:

hi， can you tell me how do you get the prompt dataset when you run the code about the stage3?

Actually all data are just text format and you have to use your own domain data to train your own domain models. You may find the format in my PR codes. The dataset code is the easy_dataset.py.

**Comment by HaixHan** [comment]
Hello， my friend！ I just want to tell you that  I know the reason about the question i asked you yesterday! By the way, Is right your github code about collosalAI in stage 2? 



（你github上关于colossalAI的stage2是否跑通了？，哈哈哈哈哈哈哈，中国人）










At 2023-04-06 10:34:05, "YY Lin" ***@***.***> wrote:

hi， can you tell me how do you get the prompt dataset when you run the code about the stage3?

Actually all data are just text format and you have to use your own domain data to train your own domain models. You may find the format in my PR codes. The dataset code is the easy_dataset.py.
